docs/spec/manifest-v2-2.md

```diff
@@ -0,0 +1,109 @@
+# Distribution Content Manifests
+
+A Content Manifest is a simple JSON file which contains general fields that
+describe an object and its dependencies. The goal is for these manifests to
+describe a reference to any application data, optional metadata or labels, and
+to reference any dependent objects in a content-addressable and verifiable way.
+These manifests can also be stored in a repository and referenced by digest.
+
+## Field Descriptions
+
+- **`schemaVersion`** *int*
+
+    Specifies the schema version of the manifest as an integer. This document 
+    describes version `2` only.
+
+- **`target`** *object*
+
+    The target field may describe *any* object stored in a repository, allowing
```

Comment:
It's not that they are no longer allowed, but that they were not necessary. This latest update tries to be more focused and explicit that these aren't container image manifests but simply generic blob reference manifests which describes a target object and dependent objects referenced in the same repository.

The main thing to take from this change (and that I should probably be more explicit about in the document) is that the registry will ensure that objects that these generic manifests refer to (target and dependencies) *always* exist in the repository. This way it can use manifests and tags to keep track of what objects in the repository are being used, and we can also implement simple garbage collection in this way: if any blob or manifest is not referenced by either a tag or manifest then it is safe to delete. A manifest could refer to an object in another repository but then we would have no way of ensuring that that object can't be deleted from the remote repository. At the very least, we could only ensure that it exists initially.

I know you're concerned about layer federation for Docker container images so, using that case as an example, the target application data would be a docker image manifest (which would look more like what was previously in this specification with all of the application-specific metadata, runtime config, rootfs layers, etc). If your container image depended on another base image, then that base image would be in another repository and your image would refer to the base image by its full name and tag/digest reference. Docker would be responsible for fetching the base image, and it would be your job as the image maintainer to ensure that the base image exists.